### PR TITLE
BPS-626 Added repository layer for the report

### PR DESCRIPTION
**Pull request by aliveni**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/BPS-626

### Change description ###
Added repository layer to get the letters count summary report

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/sendletter/entity/LettersCountSummaryRepository.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+public interface LettersCountSummaryRepository extends JpaRepository<Letter, UUID> {
+
+    @Query(
+        nativeQuery = true,
+        value = "SELECT COUNT(1) as uploaded, service FROM letters\n"
+            + "  WHERE \n"
+            + "     sent_to_print_at >= :dateFrom\n"
+            + "         AND \n"
+            + "     sent_to_print_at <=  :dateTo\n"
```
Comment (rkondratowicz) [inline src/main/java/uk/gov/hmcts/reform/sendletter/entity/LettersCountSummaryRepository.java]: let's stick to same formatting as always:
```
WHERE sent_to_print_at >= :dateFrom
AND   sent_to_print_at <= :dateTo
```
```

File: src/integrationTest/java/uk/gov/hmcts/reform/sendletter/entity/LettersCountSummaryRepositoryTest.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+        repository.save(letterEntity("b.service", date.minusDays(2)));
+
+        List<Letter> letters = Lists.newArrayList(repository.findAll());
+        assertThat(letters.size()).isEqualTo(2);
+
+        //when
+        Date dateFrom = new SimpleDateFormat("yyyy-MM-dd HH:mm").parse("2019-04-24 17:00");
+        Date dateTo = new SimpleDateFormat("yyyy-MM-dd HH:mm").parse("2019-04-25 16:00");
+
+        List<ServiceLettersCountSummary> result = repository.countByDate(dateFrom, dateTo);
```
Comment (rkondratowicz) [inline src/integrationTest/java/uk/gov/hmcts/reform/sendletter/entity/LettersCountSummaryRepositoryTest.java]: can repo also take `LocalDateTime`?

Comment (aliveni) [reply src/integrationTest/java/uk/gov/hmcts/reform/sendletter/entity/LettersCountSummaryRepositoryTest.java]: yeah, it is working with `LocalDateTime`. I'll update it

File: src/main/java/uk/gov/hmcts/reform/sendletter/entity/LettersCountSummaryRepository.java (hunk 1 above)
Comment (doncem) [reply src/main/java/uk/gov/hmcts/reform/sendletter/entity/LettersCountSummaryRepository.java]: should be
```sql
WHERE sent_to_print_at >= :dateFrom
  AND sent_to_print_at <= :dateTo
```
but nvm

Comment (aliveni) [reply src/main/java/uk/gov/hmcts/reform/sendletter/entity/LettersCountSummaryRepository.java]: @doncem updated the code

**Comment by aliveni** [comment]
> Same on `LocalDateTime`. Make sure you're building it with `TimeZones` which is present in `util` package

@doncem Updated the pr to format the dates with ZoneId

**Pull request merged**